### ISSUE TITLE
Adam optimizer without first moment estimate, for less vram

### ISSUE THREAD

**Issue by pyphan1** [pytorch/pytorch#138248]
### 🚀 The feature, motivation and pitch

Many GANs like StyleGan use the Adam optimizer with betas=(0.0, 0.999), this means that the first moment estimate is disabled but it still uses a lot of vram for it, is it possible to add an option to stop this from happening and reduce vram usage?

**Comment by janeyx99** [comment]
Is the request to not allocate state for momentum if the first beta is 0 and to use grad as the momentum in the calculations?

If so, that should be doable!

**Comment by pyphan1** [comment]
@janeyx99 yes that's exactly it! it would solve the unnecessary high vram usage when training GANs

**Comment by janeyx99** [comment]
Yes we should do this! Marking as actionable and will try to get to in the next few weeks but please feel free to open a PR earlier than that.

**Comment by youssef62** [comment]
Hello @janeyx99, I am interested in solving this issue. Would you mind if I had a try at it?

**Comment by janeyx99** [comment]
go for it! i’m on vacation for the next 3 weeks but will review then!
